src/responder/pam/pam_prompting_config.c

```diff
@@ -346,10 +347,67 @@ errno_t pam_eval_prompting_config(struct pam_ctx *pctx, struct pam_data *pd)
         }
     }
 
+    *_pc_list = pc_list;
     ret = EOK;
 done:
     free(resp_data);
-    pc_list_free(pc_list);
+
```

Comment:
Hi,

I think in case of an error `pc_list` is leaking. I would suggest to add something like
```
    if (ret != EOK) {
        pc_list_free(pc_list);
    }
````

bye,
Sumit

Reply:
Yeah, you are right. This is fixed

Reply:
Thanks.